### DISCUSSION TITLE
ilogtail以纯插件模式启动和编译c++核心部分启动功能有多少区别？

### DISCUSSION THREAD

**Discussion by feiniao233** [alibaba/ilogtail#308] [Help] [answered]
麻烦问一下ilogtail以纯插件模式启动和编译c++核心部分启动功能有多少区别？后续更新的热更新配置文件及统一配置中心功能是否只是c++核心部分提供的功能呢？

**Comment by messixukejia** [comment, marked as answer]
纯插件模式并不是iLogtail支持的推荐工作模式，主要是用来golang 部分代码快速开发的模式。

C++跟Golang一起才是iLogtail的推荐工作模式。配置管理、文件采集都是由C++来进行的，后续统一的配置中心也是由C++部分进行服务端交互。


ilogtail生产环境如下：
├── ilogtail (主程序）
├── libPluginAdapter.so（插件接口）
└── libPluginBase.so (插件lib）

具体开发可参见：https://zhuanlan.zhihu.com/p/552057171

**Reply by feiniao233** [reply]
请问c++部分对于windows平台的支持有计划吗？或者说有什么痛点难点导致现在暂不支持windows平台？

**Reply by messixukejia** [reply]
Windows企业版是支持的。主要是编译依赖复杂些，暂时没有对外开放。之前关于之前操作系统的一些讨论https://github.com/alibaba/ilogtail/discussions/251#discussioncomment-3314449

近期应该不会把重点放在Windows的开源支持上，是否可以先描述下你们的需求，我们好评估下。 cc @yyuuttaaoo

**Reply by yiiewang** [reply]
Windows 部分可以开源出来，自行编译吗？

**Reply by yiiewang** [reply]
快速编译和 docker 编译的 libPluginAdapter.so 有什么区别吗？

**Reply by messixukejia** [reply]
> Windows 部分可以开源出来，自行编译吗？

可以的。可以加入我们钉钉群留下言，我们单独会跟你交流下。

> 快速编译和 docker 编译的 libPluginAdapter.so 有什么区别吗？

可以看下这个文章https://zhuanlan.zhihu.com/p/552057171

**Comment by codingdancer** [comment]
ilogtail  以C-shared模式编译  动态更新配置 有啥方式吗

**Reply by messixukejia** [reply]
https://github.com/alibaba/ilogtail/blob/main/docs/cn/awesome-ilogtail/getting-started.md

详见配置热加载

**Comment by codingdancer** [comment]
大佬，帮忙看看， 按以下配置 部署  ，程序正常启动，但是没有收到kafka 消息，  input的kafka 中是有产生消息的（已验证）。谢谢

apiVersion: v1
kind: Namespace
metadata:
  name: ilogtail


apiVersion: v1
kind: ConfigMap
metadata:
  name: ilogtail-user-cm
  namespace: ilogtail
data:
  kafka_kakfa.yaml: |
    enable: true
    inputs:
      - Type: service_kafka
        Version: 2.1.1
        Brokers: 
          - 10.129.168.144:9092
        ConsumerGroup: skyeyeResult
        Topic:
          - risk_qrs_to_dataexchange
        ClientID: ilogtail
    flushers:   
      - Type: flusher_kafka
        Brokers: 
          - 10.129.168.144:9092
        Topic: bmw_bms_collector_rsk



apiVersion: apps/v1
kind: Deployment
metadata:
  name: ilogtail-dm
  namespace: ilogtail-ns
  labels:
    k8s-app: ilogtail-dm
spec:
  replicas: 1
  selector:
    matchLabels:
      k8s-app: ilogtail-dm
  template:
    metadata:
      labels:
        k8s-app: ilogtail-dm
    spec:
      containers:
        - image: 'sls-opensource-registry.cn-shanghai.cr.aliyuncs.com/ilogtail-community-edition/ilogtail:latest'
          name: ilogtail
          env:
          	- name: ALIYUN_LOG_ENV_TAGS # add log tags from env
        	  value: _node_name_|_node_ip_
            - name: _node_name_
              valueFrom:
          		fieldRef:
                  apiVersion: v1
         		  fieldPath: spec.nodeName
          	- name: mem_usage_limit
              value: "512"
          resources:
            limits:
              cpu: 1000m
              memory: 1Gi
            requests:
              cpu: 400m
              memory: 384Mi
          volumeMounts:
            - mountPath: /var/run # for container runtime socket
              name: run
            - mountPath: /logtail_host # for log access on the node
              mountPropagation: HostToContainer
              name: root
              readOnly: true
            - mountPath: /usr/local/ilogtail/checkpoint # for checkpoint between container restart
              name: checkpoint
            - mountPath: /usr/local/ilogtail/user_yaml_config.d # mount config dir
              name: user-config
              readOnly: true
      volumes:
        - hostPath:
            path: /var/run
            type: Directory
          name: run
        - hostPath:
            path: /
            type: Directory
          name: root
        - hostPath:
            path: /etc/ilogtail-ilogtail-dm/checkpoint
            type: DirectoryOrCreate
          name: checkpoint
        - configMap:
            defaultMode: 420
            name: ilogtail-user-cm
          name: user-config
